crates/factor-outbound-networking/src/config.rs

```diff
@@ -562,6 +569,25 @@ mod test {
     use super::*;
     use std::net::{Ipv4Addr, Ipv6Addr};
 
+    #[test]
+    fn outbound_url_handles_at_in_paths() {
+        let url = "https://example.com/file@0.1.0.json";
+        let url = OutboundUrl::parse(url, "https").expect("should have parsed url");
+        assert_eq!("example.com", url.host);
+
+        let url = "https://user:password@example.com/file@0.1.0.json";
+        let url = OutboundUrl::parse(url, "https").expect("should have parsed url");
+        assert_eq!("example.com", url.host);
+
+        let url = "https://user:pass#word@example.com/file@0.1.0.json";
```

Comment:
This isn't a valid url, fwiw (can't have an unescaped `#`)...

Reply:
@lann Hmm, this is the example that Ryan was trying to address though (https://github.com/fermyon/spin/pull/2116).  If this can't happen then why do we need all the `@` shenanigans?

Reply:
WELL WHO REVIEWED THAT PR...oh...

I have no explanation :shrug:

Reply:
Ah sure, this was mostly in the context of "throw away this code and use some sort of off-the-shelf URL parser".